Cubical/Foundations/GroupoidLaws.agda

```diff
@@ -0,0 +1,95 @@
+{-
+
+This file proves the higher groupoid structure of types
+
+-}
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Foundations.GroupoidLaws where
+
+open import Cubical.Core.Primitives
+open import Cubical.Core.Prelude
+
+private
+  variable
+    ℓ : Level
+    A : Set ℓ
+
+-- some useful notation (move to prelude?)
+
+_·_ :  {x y z : A} → (x ≡ y) → (y ≡ z) → (x ≡ z)
+x≡y · y≡z = compPath x≡y y≡z 
+
+infixr 30 _·_
+
+_⁻¹ : {x y : A} → (x ≡ y) → (y ≡ x)
+x≡y ⁻¹ = sym x≡y
+
+-- homogeneous groupoid laws
+
+symRefl : (x : A) →
+  Path (Path A x x) refl (refl ⁻¹)
+symRefl x i = refl 
+
+compPathRefl : (x : A) →
+  Path (Path A x x) refl (refl · refl)
+compPathRefl x j i =
+  hfill (λ j → λ { (i = i0) → x
+                  ; (i = i1) → x }) (inc x) j
+
+symInvo : (x y : A) (p : x ≡ y) →
+     Path (Path A x y) p ((p ⁻¹)⁻¹)
+symInvo x y p = refl
+
+rUnit : (x y : A) (p : x ≡ y) →
```

Comment:
Why no prove compPathRefl using this?

Reply:
Oh, I was being silly. Actually compPathRefl was just an exercise, I forgot to go back to it after I proved rUnit!